### PR TITLE
Barkha_Assignment_17_65min

### PR DESCRIPTION
**Pull request by BarkhaPJun20**
1) Count Digits,Letters, UpperCase characters, LowerCase characters, Special characters from predefined string.

Input: 1rRpd3F9#K(E
Output : 
Total digit: 3
Total Letters: 7
Total Uppercase: 4
Total Lowercase: 3
Total Special char: 2

2) Write a program which print first and last index of every characters only if character has multiple occurrence.

Input : technocredits
Output :
t -> first index = 0 Last index = 11
e -> first index = 1 Last index = 8
c -> first index = 2 Last index = 6

Hint : To find first index of character --> indexOf() method
            To find last index of character --> lastIndexOf() method.
Program 1:
image
Program 2:
image

### CODE REVIEW

File: src/barkhaPJun20/CountDigitsLettersUpperLowerSpecial.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+		int count=0;
+		for(int index=0; index<name.length(); index++) {
+			char ch=name.charAt(index);
+			if (Character.isLowerCase(ch))
+					count++;
+		}
+		System.out.println("Number of LOWER CASE LETTERS in Predefined String are: "+count);
+		
+	}
+	void countSpecialCharInString(String name) {
```
Comment (KrishnaKTechnocredits) [inline src/barkhaPJun20/CountDigitsLettersUpperLowerSpecial.java]: Barkha, my expectation is to write a single method in which all conditions should be checked. and in that method only expected output should be printed.

File: src/barkhaPJun20/FirstLastIndexOfMultiOccurence.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
+package barkhaPJun20;
+
+public class FirstLastIndexOfMultiOccurence {
+	
+	void findFirstLastIndex(String name) {
+		for(int outerIndex=0; outerIndex<name.length(); outerIndex++) {
```
Comment (KrishnaKTechnocredits) [inline src/barkhaPJun20/FirstLastIndexOfMultiOccurence.java]: No need to give a variable name as outerIndex, index is fine.

File: src/barkhaPJun20/CountDigitsLettersUpperLowerSpecial.java (hunk 1 above)
Comment (BarkhaPJun20) [reply src/barkhaPJun20/CountDigitsLettersUpperLowerSpecial.java]: Wrote in if else loop...please review.

File: src/barkhaPJun20/FirstLastIndexOfMultiOccurence.java (hunk 2 above)
Comment (BarkhaPJun20) [reply src/barkhaPJun20/FirstLastIndexOfMultiOccurence.java]: Done.
image

File: src/barkhaPJun20/CountDigitsLettersUpperLowerSpecial.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+		int countLowerCase=0;
+		int countSpecialChar=0;
+		for(int index=0; index<name.length(); index++) {
+			char ch=name.charAt(index);
+			
+			if (Character.isDigit(ch)) {
+				countDigit++;
+			} else if (Character.isLetter(ch)) {
+				countLetters++;
+				if (Character.isUpperCase(ch)) {
```
Comment (KrishnaKTechnocredits) [inline src/barkhaPJun20/CountDigitsLettersUpperLowerSpecial.java]: You can achieve expected output without using countLetters variable and  else if (Character.isLetter(ch)) block. Think about it.

File: src/barkhaPJun20/FirstLastIndexOfMultiOccurence.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+public class FirstLastIndexOfMultiOccurence {
+	
+	void findFirstLastIndex(String name) {
+		for(int index=0; index<name.length(); index++) {
+			char ch=name.charAt(index);
+			
+			if(index==name.indexOf(ch)) {
+				
+				if (name.indexOf(ch)!=name.lastIndexOf(ch))
+			System.out.println(ch+""+"(" +name +") -> First Index in String is: "+name.indexOf(ch) +" & Last Index in String is: "+name.lastIndexOf(ch));
```
Comment (KrishnaKTechnocredits) [inline src/barkhaPJun20/FirstLastIndexOfMultiOccurence.java:12]: Perfect code.

File: src/barkhaPJun20/CountDigitsLettersUpperLowerSpecial.java (hunk 3 above)
Comment (BarkhaPJun20) [reply src/barkhaPJun20/CountDigitsLettersUpperLowerSpecial.java]: added two counts (countUpperCase+countLowerCase) to get overall count of letters(countLetters).

File: src/barkhaPJun20/CountDigitsLettersUpperLowerSpecial.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+				else if(Character.isLowerCase(ch)) {
+					countLowerCase++;
+				}
+				else
+					countSpecialChar++;
+			
+			}
+		System.out.println("The DIGITS present in "+name +" are: "+countDigit);
+		//System.out.println("The LETTERS present in "+name +" are: "+countLetters);	
+		System.out.println("The LETTERS present in "+name +" are: "+(countUpperCase+countLowerCase));
```
Comment (KrishnaKTechnocredits) [inline src/barkhaPJun20/CountDigitsLettersUpperLowerSpecial.java:29]: Good job.

**Review by KrishnaKTechnocredits — APPROVED** [review]

**Pull request merged**